### ISSUE TITLE
BRepBuilderAPI_ModifyShape API change in OCE 0.18 / OCCT 6.9

### ISSUE THREAD

**Issue by XB32Z** [jeromerobert/jCAE#13]
Hi, 
I tried to compile occjava using the README instruction but I got some errors. I used:
- oce master
- Ubuntu 16 
- gcc-5.4
- swig 3.0.8

I have the following error: 
`jCAE/occjava/src-java/org/jcae/opencascade/jni/OccJavaJAVA_wrap.cxx: In function ‘jlong Java_org_jcae_opencascade_jni_OccJavaJNI_BRepBuilderAPI_1ModifyShape_1modifiedShape(JNIEnv*, jclass, jlong, jobject, jlong, jobject)’:
/projects/jCAE/occjava/src-java/org/jcae/opencascade/jni/OccJavaJAVA_wrap.cxx:5123:115: error: taking address of temporary [-fpermissive]
   result = (TopoDS_Shape *) &((BRepBuilderAPI_ModifyShape const *)arg1)->ModifiedShape((TopoDS_Shape const &)*arg2);
`
Thanks for the help,

**Comment by jeromerobert** [comment]
This is because of this change in the OCCT 6.9 API / OCE 0.18:

```patch
---------------------- inc/BRepBuilderAPI_ModifyShape.hxx ----------------------
index 8511b6c7c..e3af84bb1 100644
@@ -55,7 +55,7 @@ public:
   //! transformation has been applied. Raises NoSuchObject from Standard
   //! if S is not the initial shape or a sub-shape
   //! of the initial shape.
-  Standard_EXPORT virtual  const  TopoDS_Shape& ModifiedShape (const TopoDS_Shape& S)  const;
+  Standard_EXPORT virtual   TopoDS_Shape ModifiedShape (const TopoDS_Shape& S)  const;
```
Until I fix it, only OCE <= 0.17 / OCCT <= 6.8 will work.

**Comment by jeromerobert** [comment]
The fix should be similar to what I did for `gp_Trsf::SetValues`:
* https://github.com/jeromerobert/jCAE/blob/master/occjava/src/gp.i#L31
* https://github.com/jeromerobert/jCAE/blob/master/occjava/src/gp.i#L406
